modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/CodeArea.java

```diff
@@ -0,0 +1,425 @@
+        CodeTextModel m = codeModel();
+        return (m == null) ? null : m.getDecorator();
+    }
+
+    /**
+     * Determines whether to show line numbers.
+     * @return the line numbers enabled property
+     * @defaultValue false
+     */
+    // TODO should there be a way to customize the line number component? createLineNumberDecorator() ?
+    // TODO should this be a styleable property?
+    public final BooleanProperty lineNumbersEnabledProperty() {
+        if (lineNumbers == null) {
+            lineNumbers = new SimpleBooleanProperty() {
+                @Override
+                protected void invalidated() {
+                    LineNumberDecorator d;
+                    if (get()) {
+                        // TODO create line number decorator method?
+                        d = new LineNumberDecorator() {
+                            @Override
+                            public Node getNode(int ix, boolean forMeasurement) {
+                                Node n = super.getNode(ix, forMeasurement);
+                                if (n instanceof Labeled t) {
+                                    t.fontProperty().bind(fontProperty());
+                                }
+                                return n;
+                            }
+                        };
+                    } else {
+                        d = null;
+                    }
+                    setLeftDecorator(d);
+                }
+            };
+        }
+        return lineNumbers;
+    }
+
+    public final boolean isLineNumbersEnabled() {
+        return lineNumbers == null ? false : lineNumbers.get();
+    }
+
+    public final void setLineNumbersEnabled(boolean on) {
+        lineNumbersEnabledProperty().set(on);
+    }
+
+    /**
+     * The size of a tab stop in spaces.
+     * Values less than 1 are treated as 1.
+     * @return the tab size property
+     * @defaultValue 8
+     */
+    public final IntegerProperty tabSizeProperty() {
+        if (tabSize == null) {
+            tabSize = new StyleableIntegerProperty(Params.DEFAULT_TAB_SIZE) {
+                @Override
+                public Object getBean() {
+                    return CodeArea.this;
+                }
+
+                @Override
+                public String getName() {
+                    return "tabSize";
+                }
+
+                @Override
+                public CssMetaData getCssMetaData() {
+                    return StyleableProperties.TAB_SIZE;
+                }
+            };
+        }
+        return tabSize;
+    }
+
+    public final int getTabSize() {
+        return tabSize == null ? Params.DEFAULT_TAB_SIZE : tabSize.get();
+    }
+
+    public final void setTabSize(int spaces) {
+        tabSizeProperty().set(spaces);
+    }
+
+    /**
+     * The font to use for text in the {@code CodeArea}.
+     * @return the font property
+     * @defaultValue the Monospaced font with size 12.0 px
```

Comment:
Would a monospaced font the size of the default system font,  `1em`, be a better choice or do you think hard-coding 12 point is best?

Reply:
It's created with `Font.font("Monospaced", 12.0);` but perhaps it should use "the default size" whatever that means

`Font.font("Monospaced", -1);`